examples/ensemble/plot_stack_predictors.py

```diff
@@ -16,42 +16,126 @@
-    ax.legend([extra], [scores], loc='upper left')
-    title = title + '\n Evaluation in {:.2f} seconds'.format(elapsed_time)
-    ax.set_title(title)
+def load_ames_housing():
+    df = fetch_openml(data_id=42165, as_frame=True)
+    X = df.data
+    y = df.target
+
+    features = ['YrSold', 'HeatingQC', 'Street', 'YearRemodAdd', 'Heating',
+                'MasVnrType', 'BsmtUnfSF', 'Foundation', 'MasVnrArea',
+                'MSSubClass', 'ExterQual', 'Condition2', 'GarageCars',
+                'GarageType', 'OverallQual', 'TotalBsmtSF', 'BsmtFinSF1',
+                'HouseStyle', 'MiscFeature', 'MoSold']
+
+    X = X[features]
+    return X, y
+
+
+X, y = load_ames_housing()
+
+
+###############################################################################
+# Make pipeline to preprocess the data
+###############################################################################
+#
+# Before we can use Ames dataset we still need to do some preprocessing.
+# First, the dataset has many missing values. To impute them, we will exchange
+# categorical missing values with the new category 'missing' while the
+# numerical missing values with the 'mean' of the column. We will also encode
+# the categories with either :class:`sklearn.preprocessing.OneHotEncoder
+# <sklearn.preprocessing.OneHotEncoder>` or
+# :class:`sklearn.preprocessing.OrdinalEncoder
+# <sklearn.preprocessing.OrdinalEncoder> depending for which type of model we
+# will use them (linear or non-linear model). To falicitate this preprocessing
+# we will make two pipelines.
+# You can skip this section if your data is ready to use and does
+# not need preprocessing
+
+
+from sklearn.compose import make_column_transformer
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import OrdinalEncoder
+from sklearn.preprocessing import StandardScaler
+
+
+cat_cols = X.columns[X.dtypes == 'O']
+num_cols = X.columns[X.dtypes == 'float64']
+
+categories = [
+    X[column].unique() for column in X[cat_cols]]
+
+
+for cat in categories:
+    cat[cat == None] = 'missing'
+
+cat_proc_nlin = make_pipeline(
+    SimpleImputer(missing_values=None, strategy='constant',
+                  fill_value='missing'),
+    OrdinalEncoder(categories=categories)
+    )
+
+num_proc_nlin = make_pipeline(SimpleImputer(strategy='mean'))
+
+cat_proc_lin = make_pipeline(
+    SimpleImputer(missing_values=None,
+                  strategy='constant',
+                  fill_value='missing'),
+    OneHotEncoder(categories=categories)
+)
+
+num_proc_lin = make_pipeline(
+    SimpleImputer(strategy='mean'),
+    StandardScaler()
+)
+
+# transformation to use for non-linear estimators
+processor_nlin = make_column_transformer(
+                              (cat_proc_nlin, cat_cols),
+                              (num_proc_nlin, num_cols),
+                              remainder='passthrough')
```

Comment:
nit: use only one level of indentation (4 spaces) on a new lines after an open parenthesis.